tests/checks/string.fish

```diff
@@ -42,6 +42,21 @@ string length "hello, world"
 string length -q ""; and echo not zero length; or echo zero length
 # CHECK: zero length
 
+string pad foo
+# CHECK: foo
```

Comment:
Can I reuse `aiter` in this case?

I'm not sure if I can reuse existing one or better to create a new one?

Reply:
I have addressed this in separate commit (+ test for that).

Reply:
The iterator not currently reusable. Creating a new `arg_iterator_t` could work when arguments are given on the commandline.

However when something like `echo foo\nbar | string pad` is run, then the iterator would consume the input, so that won't work anymore.

You can create a temporary `std:;vector<wcstring> arguments;` where you keep all input arguments. Then it's possible to measure the maximum width and in a second step output them.

If an explicit width is given, the vector is not needed, but optimizations can be done later.

Reply:
Ok, I see what you mean with `echo`, I have made changes to keep the inputs in separate vector.